Source/WebCore/page/Page.cpp

```diff
@@ -1106,7 +1106,15 @@ bool Page::shouldBuildInteractionRegions() const
 {
     return m_settings->interactionRegionsEnabled();
 }
-#endif
+
+void Page::setInteractionRegionsEnabled(bool enable)
+{
+    bool needsUpdate = enable && !shouldBuildInteractionRegions();
+    m_settings->setInteractionRegionsEnabled(enable);
```

Comment:
This means the Setting and WebPreference get out of sync, right? (and then if something else mutates a WebPreference we'll overwrite it?) I think you could avoid all the new IPC messages *and* keep them in sync if you just mutate the WebPreference (as suggested in https://github.com/WebKit/WebKit/pull/2684#discussion_r928069394).

Also, then you could move the SPI to WKPreferences(Private).h. A lot of the patch dissolves.

But maybe I am missing a good reason it's on the view instead? Maybe it's about WKPreference-sharing?

Reply:
I'll take another look at this. If I use WKPreferences, I can mutate that value during the lifetime of the web view, for instance when the preview web view is selected, and expect that to work and rebuild the regions? I'm a little unfamiliar with the interaction but I'll take a look

Reply:
WKPreferences push state down to WebCore::Settings, and update dynamically, yes.

Reply:
(Via WebKit::WebPreferences)

Reply:
Oh nice, yes that sounds like a better approach. I’ll do that

Reply:
@hortont424 and I discussed offline and it seems like this is non-trivial to implement via WKPreferences

Reply:
(trivial in WebKit, but doesn't work out for clients)